commit 90944c5323c3
Author: Martin Stransky <stransky@redhat.com>
Date:   Mon Apr 9 13:36:03 2018 +0200

    Bug 1452041 - nsClipboard::HasDataMatchingFlavors(): return immediately when there's no clipboard content, r=jhorak a=jcristau
    
    MozReview-Commit-ID: CTz0tRr3p57
    
    --HG--
    extra : rebase_source : 35453506a58e698d8f0b6e9681fb87da8511eb9c
    extra : source : 4b841d1c7ec0513eb080fb033f54bde12ccb7ddc
---
 widget/gtk/nsClipboard.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git widget/gtk/nsClipboard.cpp widget/gtk/nsClipboard.cpp
index 36e8d6ad5cee..61d3ef53ca98 100644
--- widget/gtk/nsClipboard.cpp
+++ widget/gtk/nsClipboard.cpp
@@ -390,6 +390,8 @@ nsClipboard::HasDataMatchingFlavors(const char** aFlavorList, uint32_t aLength,
 
   int targetNums;
   GdkAtom* targets = mContext->GetTargets(aWhichClipboard, &targetNums);
+  if (!targets)
+      return NS_OK;
 
   // Walk through the provided types and try to match it to a
   // provided type.
